crates/storage/src/lazy/mapping.rs

```diff
@@ -164,6 +180,45 @@ const _: () = {
 mod tests {
     use super::*;
 
+    /// A dummy type which `REQUIRES_DEEP_CLEAN_UP`
+    #[derive(PartialEq, Debug, scale::Encode, scale::Decode)]
+    struct DeepClean<T>(T);
+
+    impl<T> SpreadLayout for DeepClean<T>
+    where
+        T: PackedLayout,
+    {
+        const FOOTPRINT: u64 = 1;
+        const REQUIRES_DEEP_CLEAN_UP: bool = true;
+
+        fn pull_spread(ptr: &mut KeyPtr) -> Self {
+            DeepClean(<T as SpreadLayout>::pull_spread(ptr))
+        }
+
+        fn push_spread(&self, ptr: &mut KeyPtr) {
+            <T as SpreadLayout>::push_spread(&self.0, ptr)
+        }
+
+        fn clear_spread(&self, ptr: &mut KeyPtr) {
+            <T as SpreadLayout>::clear_spread(&self.0, ptr)
+        }
+    }
+
+    impl<T> PackedLayout for DeepClean<T>
+    where
+        T: PackedLayout + scale::EncodeLike<T>,
+    {
+        fn pull_packed(&mut self, at: &Key) {
+            <T as PackedLayout>::pull_packed(&mut self.0, at);
+        }
+        fn push_packed(&self, at: &Key) {
+            <T as PackedLayout>::push_packed(&self.0, at);
+        }
+        fn clear_packed(&self, at: &Key) {
+            <T as PackedLayout>::clear_packed(&self.0, at);
+        }
+    }
```

Comment:
Hmm, I thought we have a derive macro for this somewhere.

Reply:
So we have `#[derive(SpreadLayout, PackedLayout)]` (see the `multisig` example), can you check if you can get rid of the two impls here with the macro?

Reply:
that won't be possible for the `SpreadLayout` since it heavily diverges from the derive macro automated implementation.
For the `PackedLayout` I am not sure but I would personally keep it without derive because `ink_storage` is a frequently compiled crate and so far we did a good job in not using the derive macros from `ink_storage_derive` within the `ink_storage` crate for the reason of compilation time reduction.

Reply:
Ah I didn't even think of using derive macros for this, I'll give it a go

Reply:
Looks like I can't use the `derive` macros within this crate 🤷 Either way, we'd still need at least one manual implementation since we want to set `REQUIRES_DEEP_CLEAN_UP = true`

Reply:
The reason is comoilation time and also that most types defined in ink_storage cannot use derives anyways since they are forming the basic building ground. Therefore we do not make use of storage derives in the crate itself.